pkg/engine/task/task_delete.go

```diff
@@ -0,0 +1,53 @@
+package task
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DeleteTask will delete a set of given resources from the cluster. See Run method for more details.
+type DeleteTask struct {
+	Name      string
+	Resources []string
+}
+
+// Run method for the DeleteTask. Given the task context, it renders the templates using context parameters
+// creates runtime objects and kustomizes them, and finally removes them using the controller client.
+func (dt DeleteTask) Run(ctx Context) (bool, error) {
+	// 1. - Render task templates -
+	rendered, err := render(dt.Resources, ctx.Templates, ctx.Parameters, ctx.Meta)
+	if err != nil {
+		return false, fmt.Errorf("%wfailed to render task resources: %v", ErrFatalExecution, err)
+	}
+
+	// 2. - Kustomize them with metadata -
+	kustomized, err := kustomize(rendered, ctx.Meta, ctx.Enhancer)
```

Comment:
why do we need to kustomize them? is it because of the name decoration we do? sounds about right, just worth documenting since it looks a little odd on first look.

Reply:
I'm not sure if have to. This logic got adopted from _master_ and I didn't question it. I'd also rather tackle this in a separate PR.

Reply:
yeah I actually think we don't have to. To be honest, I think that the DeleteTask should not even accept resources, specifying the identifier of that resource should be enough but it would probably need some API design... having resources there is easier now :)